sklearn/meson.build

```diff
@@ -60,6 +60,25 @@ np_dep = declare_dependency(include_directories: inc_np)
 openmp_dep = dependency('OpenMP', language: 'c', required: false)
 
 if not openmp_dep.found()
+  warn_about_missing_openmp = true
+  # On Apple Clang avoid a misleading warning if compiler variables are set.
+  # See https://github.com/scikit-learn/scikit-learn/issues/28710 for more
+  # details
```

Comment:
Maybe make it more explicit that it's temporary until meson fixes it.

Reply:
Looking at https://github.com/mesonbuild/meson/issues/7435#issuecomment-2052229398 I think this may not be that trivial to fix in Meson:

```
How exactly to handle the rest of it is more complicated. We could inject user supplied arguments into the dependencies, although that seems fragile and is going to be a big project to make sure we're doing that consistently.
```

Reply:
I mention this + link to Meson issue